test/integration/hotrestart_test.sh

```diff
@@ -48,23 +120,22 @@ fi
 TEST_INDEX=0
 for HOT_RESTART_JSON in "${JSON_TEST_ARRAY[@]}"
 do
-  # Test validation.
   # TODO(jun03): instead of setting the base-id, the validate server should use the nop hot restart
-  "${ENVOY_BIN}" -c "${HOT_RESTART_JSON}" --mode validate --service-cluster cluster \
+  start_test validation
+  check "${ENVOY_BIN}" -c "${HOT_RESTART_JSON}" --mode validate --service-cluster cluster \
       --service-node node --base-id 1
 
   # Now start the real server, hot restart it twice, and shut it all down as a basic hot restart
   # sanity test.
-  echo "Starting epoch 0"
+  start_test Starting epoch 0
   ADMIN_ADDRESS_PATH_0="${TEST_TMPDIR}"/admin.0."${TEST_INDEX}".address
-  "${ENVOY_BIN}" -c "${HOT_RESTART_JSON}" \
+  run_in_background_saving_pid "${ENVOY_BIN}" -c "${HOT_RESTART_JSON}" \
       --restart-epoch 0 --base-id 1 --service-cluster cluster --service-node node \
-      --admin-address-path "${ADMIN_ADDRESS_PATH_0}" &
+      --admin-address-path "${ADMIN_ADDRESS_PATH_0}"
 
-  FIRST_SERVER_PID=$!
-  sleep 3
```

Comment:
Without some time here (not that I love arbitrary sleep in test...), couldn't we send signals to the process before it has installed signal handlers?

Reply:
good catch; done.